src/sentry/rules/conditions/high_priority_issue.py

```diff
@@ -0,0 +1,62 @@
+from datetime import datetime
+from typing import Optional, Sequence
+
+from sentry import features
+from sentry.eventstore.models import GroupEvent
+from sentry.models.activity import Activity
+from sentry.models.group import Group
+from sentry.rules import EventState
+from sentry.rules.conditions.base import EventCondition
+from sentry.types.activity import ActivityType
+from sentry.types.condition_activity import ConditionActivity, ConditionActivityType
+
+HIGH_SEVERITY_THRESHOLD = 0.1
+
+
+class HighPriorityIssueCondition(EventCondition):
+    id = "sentry.rules.conditions.high_priority_issue.HighPriorityIssueCondition"
+    label = "Sentry marks an issue as high priority"
+
+    def is_high_severity(self, state: EventState, group: Optional[Group]) -> bool:
+        if not group:
+            return False
+
+        try:
+            severity = float(group.get_event_metadata().get("severity", ""))
+        except (KeyError, TypeError, ValueError):
+            return False
+
+        return severity >= HIGH_SEVERITY_THRESHOLD
+
+    def passes(self, event: GroupEvent, state: EventState) -> bool:
+        has_issue_priority_alerts = features.has("projects:high-priority-alerts", self.project)
+        if not has_issue_priority_alerts:
+            return False
+
+        is_high_severity = self.is_high_severity(state, event.group)
+        is_escalating = state.has_reappeared or state.has_escalated
+
+        return is_high_severity or is_escalating
+
+    def get_activity(
+        self, start: datetime, end: datetime, limit: int
+    ) -> Sequence[ConditionActivity]:
+        # reappearances are recorded as SET_UNRESOLVED with no user
+        activities = (
+            Activity.objects.filter(
+                project=self.project,
+                datetime__gte=start,
+                datetime__lt=end,
+                type=ActivityType.SET_UNRESOLVED.value,
+                user_id=None,
+            )
+            .order_by("-datetime")[:limit]
+            .values_list("group", "datetime", "data")
+        )
+
+        return [
+            ConditionActivity(
+                group_id=a[0], type=ConditionActivityType.REAPPEARED, timestamp=a[1], data=a[2]
+            )
+            for a in activities
+        ]
```

Comment:
Not sure this is comprehensive, I wonder if we should be writing an event for new escalating so you can fetch it here?

Reply:
that could work - I think we do write an activity for Escalating too so I can include for that as well 

https://github.com/getsentry/sentry/blob/8a9f0ad73d075a8fe5d94bbe913d9c885054b0bf/src/sentry/issues/escalating.py#L534

Reply:
I wonder if we should also write an activity here? https://github.com/getsentry/sentry/blob/f6fd15d3a195b11d065344ac3bb8fdc5f6153893/src/sentry/tasks/post_process.py#L1377-L1378

Reply:
yeah that makes sense to me - I can get that PR up